sandbox/apparmor/apparmor.go

```diff
@@ -617,35 +646,49 @@ func (aap *appArmorProbe) ParserFeatures() ([]string, error) {
 }
 
 func probeKernelFeatures() ([]string, error) {
-	// note that os.ReadDir() is already sorted
-	dentries, err := os.ReadDir(filepath.Join(rootPath, featuresSysPath))
+	features, err := probeKernelFeaturesInDirRecursively(filepath.Join(rootPath, featuresSysPath), "")
+	if err != nil {
+		return []string{}, err
+	}
+	if data, err := os.ReadFile(filepath.Join(rootPath, featuresSysPath, "policy", "permstable32")); err == nil {
```

Comment:
given we are revisiting this, what's the relation between permstable32 and permstable32_version, do we need to worry about the latter?

Reply:
JJ says the minimum version is 2. Technically 1 can work, but it has problems. I think we can add a check for version 2+, but the question is whether to include it in the apparmor kernel features probing, or to do a one-off check here (like we do for the socket presence).

Reply:
Per sync with Samuele, do this in the prompting support checks, rather than in the general apparmor kernel feature probing